R/commonDiscoverDistributions.R

```diff
@@ -31,14 +25,18 @@ gettextf <- function(fmt, ..., domain = NULL)  {
     jaspResults[['simdata']] <- createJaspState(sample)
     jaspResults[['simdata']]$dependOn(c("newVariableName", "simulateNow"))
 
-    if(as == "scale"){
-      .setColumnDataAsScale  (options[["newVariableName"]], sample)
-    } else if(as == "ordinal"){
-      .setColumnDataAsOrdinal(options[["newVariableName"]], sample)
-    } else{
-      .setColumnDataAsNominal(options[["newVariableName"]], sample)
+    if(options[["newVariableName"]] != "")
+    {
+      jaspResults[['sampleColumn']] <- createJaspColumn(options[["newVariableName"]])
+      
+      didItWork <- switch(as,
+      scale=  jaspResults[['sampleColumn']]$setScale(  sample),
+      ordinal=jaspResults[['sampleColumn']]$setOrdinal(sample),
+              jaspResults[['sampleColumn']]$setNominal(sample))
+
+      if(!didItWork) 
```

Comment:
I get
 
```r
This analysis terminated unexpectedly.

Error in !didItWork: invalid argument type
```

Reply:
cause you first need those PRs

Reply:
Which are "those"?

Reply:
Ah I am an idiot, you linked them already here. oops 😅